### PR TITLE
Added Dimensions.addEventListener and Dimensions.removeEventListener

### PR DESCRIPTION
**Pull request by thenewvu**
**This patch solves the following problem**

Missing of 2 official APIs [Dimensions.addEventListener](https://facebook.github.io/react-native/docs/dimensions.html#addeventlistener) and [Dimensions.removeEventListener](https://facebook.github.io/react-native/docs/dimensions.html#removeeventlistener).

**Test plan**

No.

**This pull request**

- [x] includes documentation
- [x] includes tests
- [ ] includes benchmark reports
- [x] includes an interactive example
- [ ] includes screenshots/videos

### CODE REVIEW

**Comment by RangerMauve** [comment]
This would be super useful to have! Having bugs in my app at the moment because it's missing

**Comment by thenewvu** [comment]
@necolas Excuse me, sir. Could you give some time to review this ?

**Comment by thenewvu** [comment]
@necolas Excuse me sir, could you plan some time to review this ?

File: docs/apis/Dimensions.md [hunk 1]
```diff
@@ -11,3 +11,15 @@ static **get**(dimension: string)
...
 
 Example: `const { height, width } = Dimensions.get('window')`
+
+static **addEventListener**(type, handler)
+
+Add an event handler. Supported events:
+
+* `change`: Fires when a property within the `Dimensions` object changes. The argument to the event handler is an object with `window` and `screen ` properties whose values are the same as the return values of `Dimensions.get('window')` and `Dimensions.get('screen')`, respectively.
+
+
```
Comment (necolas) [inline docs/apis/Dimensions.md]: nit: remove extra blank line

File: docs/apis/Dimensions.md [hunk 2]
```diff
@@ -11,3 +11,15 @@ static **get**(dimension: string)
...
+static **removeEventListener**(type, handler)
+
+Remove an event handler.
+
```
Comment (necolas) [inline docs/apis/Dimensions.md]: nit: remove extra blank line

File: src/apis/Dimensions/index.js [hunk 3]
```diff
@@ -44,6 +46,26 @@ export default class Dimensions {
       scale: win.devicePixelRatio || 1,
       width: win.screen.width
     };
+
+    if (canUseDOM) {
+      const change: window.Event = new window.Event(EVENT_CHANGE);
```
Comment (necolas) [inline src/apis/Dimensions/index.js]: Why did you choose to use a custom DOM event? Browser support doesn't appear to be there either

Comment (thenewvu) [inline src/apis/Dimensions/index.js]: I saw that there's no kind of an event emitter in dependencies, so custom DOM event is the only choice for me at that time. I also checked it's compatibility, it doesn't look bad. Do you have a better idea? What about adding `eventemitter3`?

Comment (RangerMauve) [inline src/apis/Dimensions/index.js]: You could probably just store an array of listener functions within Dimensions and not bother with an event emitter or the DOM. It'd be faster, too.

Comment (thenewvu) [inline src/apis/Dimensions/index.js]: oh ... okay, I will go that way.

**Comment by thenewvu** [comment]
@necolas Just added requested changes, please take time to review this.

**Comment by cloudle** [comment]
Can't wait to see this merged, @thenewvu thanks for your work.

**Comment by necolas** [comment]
Thanks

**Pull request closed without merging**